sdk/monitor/azquery/custom_client.go

```diff
@@ -27,28 +28,49 @@ type LogsClientOptions struct {
 	azcore.ClientOptions
 }
 
+// LogsClient contains the methods for the LogsClient group.
+// Don't use this type directly, use NewLogsClient() instead.
+type LogsClient struct {
+	host string
+	pl   runtime.Pipeline
+}
+
+// MetricsClient contains the methods for the Metrics group.
+// Don't use this type directly, use NewMetricsClient() instead.
+type MetricsClient struct {
+	host string
+	pl   runtime.Pipeline
+}
+
 // NewLogsClient creates a client that accesses Azure Monitor logs data.
 func NewLogsClient(credential azcore.TokenCredential, options *LogsClientOptions) *LogsClient {
 	if options == nil {
 		options = &LogsClientOptions{}
 	}
-	authPolicy := runtime.NewBearerTokenPolicy(credential, []string{"https://api.loganalytics.io/.default"}, nil)
+
+	host := cloud.AzurePublic.Services[monitorQueryLogs].Endpoint
+	if c, ok := options.Cloud.Services[monitorQueryLogs]; ok {
+		host = c.Endpoint
+	}
+	authPolicy := runtime.NewBearerTokenPolicy(credential, []string{host + "/.default"}, nil)
 	pl := runtime.NewPipeline(moduleName, version, runtime.PipelineOptions{PerRetry: []policy.Policy{authPolicy}}, &options.ClientOptions)
-	return &LogsClient{pl: pl}
+	return &LogsClient{host: host + "/v1", pl: pl}
 }
 
 // NewMetricsClient creates a client that accesses Azure Monitor metrics data.
 func NewMetricsClient(credential azcore.TokenCredential, options *MetricsClientOptions) *MetricsClient {
 	if options == nil {
 		options = &MetricsClientOptions{}
 	}
-	authPolicy := runtime.NewBearerTokenPolicy(credential, []string{"https://management.azure.com/.default"}, nil)
+	host := cloud.AzurePublic.Services[monitorQueryMetrics].Endpoint
+	if c, ok := options.Cloud.Services[monitorQueryMetrics]; ok {
+		host = c.Endpoint
+	}
+	authPolicy := runtime.NewBearerTokenPolicy(credential, []string{host + "/.default"}, nil)
```

Comment:
I'm thinking these ctors should read `ServiceConfiguration.Audience` because although the `host + /.default` pattern works for well-known clouds, I've seen scopes for Azure Stack that don't follow this pattern.

Reply:
fixed in commit [3772df1](https://github.com/Azure/azure-sdk-for-go/pull/19462/commits/3772df1fc545f3ba93ce4eb8c7ba45c582aff39d)